From a1be3551ea6d2a054fab026bf150aa23bac48638 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Tue, 31 Oct 2023 10:59:04 +0600
Subject: [PATCH 81/85] arm64: configs: vc4: enable ufs support

- enable renesas ufs driver
- build scsi disk support and ufs firmware into the kernel, to make it
  possible to boot with root on ufs

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 arch/arm64/configs/renesas-vc4_defconfig | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/configs/renesas-vc4_defconfig b/arch/arm64/configs/renesas-vc4_defconfig
index de1513aa5d08..e04ea168fe1c 100644
--- a/arch/arm64/configs/renesas-vc4_defconfig
+++ b/arch/arm64/configs/renesas-vc4_defconfig
@@ -226,6 +226,8 @@ CONFIG_PCIE_RENESAS=y
 CONFIG_PCIE_LAYERSCAPE_GEN4=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_EXTRA_FIRMWARE="r8a779f0_ufs.bin"
+CONFIG_EXTRA_FIRMWARE_DIR="firmware"
 CONFIG_FW_LOADER_USER_HELPER=y
 CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
 CONFIG_MTD=y
@@ -237,7 +239,10 @@ CONFIG_BLK_DEV_NVME=m
 CONFIG_EEPROM_AT24=y
 CONFIG_SCSI=y
 # CONFIG_SCSI_PROC_FS is not set
-CONFIG_BLK_DEV_SD=m
+CONFIG_BLK_DEV_SD=y
+CONFIG_SCSI_UFSHCD=y
+CONFIG_SCSI_UFSHCD_PLATFORM=y
+CONFIG_SCSI_UFS_RENESAS=y
 CONFIG_NETDEVICES=y
 CONFIG_MACVLAN=m
 CONFIG_MACVTAP=m
@@ -369,8 +374,6 @@ CONFIG_HWSPINLOCK=y
 CONFIG_IPMMU_VMSA=y
 CONFIG_ARCH_R8A779F0=y
 # CONFIG_RCAR_THERMAL_EMS is not set
-CONFIG_PM_DEVFREQ=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=y
 CONFIG_DEVFREQ_GOV_POWERSAVE=y
 CONFIG_DEVFREQ_GOV_USERSPACE=y
@@ -401,7 +404,6 @@ CONFIG_TMPFS=y
 CONFIG_HUGETLBFS=y
 # CONFIG_MISC_FILESYSTEMS is not set
 # CONFIG_NETWORK_FILESYSTEMS is not set
-CONFIG_NLS=y
 CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ISO8859_1=y
 CONFIG_KEYS=y
-- 
2.39.2

